src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -101,6 +102,8 @@ public String key() {
     return PROPERTY_MAPPER;
   }
 
+  private Activity currentActivity;
```

Comment:
I would also move this variable up to be with other global vars, which are around like 64.